### ISSUE TITLE
X11 Drag + Drop support: Some Issues

### ISSUE THREAD

**Issue by leezer3** [opentk/opentk#570]
Been doing some testing on the X11 drag+drop support after having the first of these reported to me.

1:
Standard Ubuntu install, en-GB locale.

Take the following filename:
```
/home/osboxes/Desktop/デスクトップ/exterior.csv
```

Attempt to drag+ drop into a gamewindow, and we get the following:
```
/home/osboxes/Desktop/%E3%83%87%E3%82%B9%E3%82%AF%E3%83%88%E3%83%83%E3%83%97/exterior.csv
```

The Unicode part of the filename has been converted into bytes and dumped into the string.
I'm not familiar with X11 and exactly what the string conversion should be returning, but this obviously doesn't work.

2:
Xubuntu 14.04 install, en-GB locale.

Crashes immediately on dropping a file into the window, with an error which states: `mono free() invalid next size (fast) c#` plus a bunch of GCC stuff/ memory addresses, which vary with each call.

I've tracked it down to the Marshal.AllocFreeHGlobal call here:
https://github.com/opentk/opentk/blob/develop/src/OpenTK/Platform/X11/X11GLNative.cs#L896
Interestingly, if I just stuff this directly after it's allocated & return from the function, it still dumps out. 

Can't explain this, nothing seems wrong to me, and the same code works just fine on the current box with Unity etc.

Again, I'm really not familiar with exactly what we're throwing around here, so no real suggestions as to what's going on......

@VPeruS  ??

**Comment by UnknownShadow200** [comment]
String looks to be url-encoded, matches up with what [SDL](https://github.com/spurious/SDL-mirror/blob/fe78534b5504327c9b8a893b9e7e1315bbd66d2e/src/video/x11/SDL_x11events.c#L267) is doing.

**Comment by Nihlus** [comment]
Closed via #571.
